packages/launch/src/utils.ts

```diff
@@ -59,3 +59,12 @@ export const getDomainProduct = (
 		source: flow,
 	},
 } );
+
+export type Product = {
+	product_id: number;
+};
```

Comment:
All products whether its a domain product or plan product has `product_id`, so at the bare minimum, I think a cart product has `product_id`.

Reply:
What do you think about creating a "base" type and let `PlanProduct` & `DomainProduct` inherit its properties? 

```
export type Product = {
    product_id: number;
    extra?: {
        source: string;
    }
}

export type PlanProduct = Product & {
    product_slug: string;
}

export type DomainProduct = Product & {
    meta: string;
    extra: {
        privacy_available: boolean;
		privacy: boolean;
    }
}
```

Reply:
Thanks! I didn't know about the intersection `&` operator!

As I was refactoring this together with https://github.com/Automattic/wp-calypso/pull/47990 I realized that `product_slug` is also part of `Product`, which we then end up in the scenario where `export type PlanProduct = Product;`.

The refactoring suggestion is great but for readability reasons I'm gonna leave things as it is for now. 

We should look into potentially reusing `type WPCOMCartItem` from `client/my-sites/checkout/composite-checkout/types/checkout-cart.ts` in the future.

Reply:
> [...]we then end up in the scenario where `export type PlanProduct = Product;`.

Hmmm, I agree that's not very nice. Even when we use `PlanProduct` as "base" and leave `Product` out, it becomes quite messy since `product_slug` is not part of `DomainProduct` and has to be optional in the base which is not very solid.

> The refactoring suggestion is great but for readability reasons, I'm gonna leave things as it is for now.

I agree for the reasons mentioned above. 

> We should look into potentially reusing `type WPCOMCartItem`[...]

Yes, for consistency reasons and solidifying our codebase this is the best solution 👍  But not as part of this PR.

Reply:
What about `ResponseCartProduct` which is already extracted to a package?
See: https://github.com/Automattic/wp-calypso/pull/47999/files#diff-e7e38dec68b1a4011df8cce1217ea7c0554561fca279540153d2216f961b97f5R6

Reply:
Using `ResponseCartProduct` seems to fit! Thank you @razvanpapadopol!